packages/rest/src/rest.ts

```diff
@@ -329,12 +360,20 @@ function makeActionName(op: Operation, name: string | undefined): string {
   return lowerCaseFirstChar(name || op.name);
 }
 
+const actionsSegmentKey = createStateSymbol("actionSegment");
+
+export function $actionSegment(context: DecoratorContext, entity: Operation, name: string) {
+  context.program.stateMap(actionsSegmentKey).set(entity, name);
+}
+
+export function getActionSegment(program: Program, entity: Type): string | undefined {
+  return program.stateMap(actionsSegmentKey).get(entity);
+}
```

Comment:
@timotheeguerin I want to _not_ export these, but when I tried to simply replace these calls with the internals... tests started failing. Any suggestions? Is there a "private" or "internal" namespace I could put these in?

Reply:
Use `setCadlNamespace` to put them in an internal sub-namespace:

```
setCadlNamespace("Internal", $actionSegment);
```

Though if we want users to use this (which I think we might) then maybe it shouldn't be in an internal namespace?

Reply:
I'd argue it's purely internal. They can already use `getAction` and `getCollectionAction`. We definitely don't want (I think) users using `@actionSegment`.